### PR TITLE
Added support for subaccounts

### LINKED ISSUE #128: Add subaccounts support

### PR DESCRIPTION
**Pull request by coldacid**
Adds a subaccounts library roughly matching the functionality and design of the templates library.

Closes #128

### CODE REVIEW

**Comment by aydrian** [comment]
Thank you for the PR. Just in time for the 1.3 release slated for tomorrow. We'll get this reviewed ASAP.

**Comment by aydrian** [comment]
So it looks good, but we are changes the way we are doing some things. Just put everything you need to for create and update in `options`, no need to create `options.subaccount`. We'll be changing this for the other libraries in 2.0.0. Might as well start now for this resource. You can see the same in Relay Webhooks. Also, you can remove options for find and have it take just the `subaccount_id`.
We are also working to remove the `toApiFormat` function. See how we are doing it with Relay Webhooks. We have found that one function does not rule them all.

**Comment by coldacid** [comment]
Alright, I was able to get this done today after all. The subaccount library now matches the new API style as presented by the Relay Webhooks library.

File: test/spec/subaccounts.speq.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
+var chai = require('chai')
```
Comment (bizob2828) [inline test/spec/subaccounts.speq.js]: This file is misspelled. Must be subaccounts.spec.js

Comment (coldacid) [inline test/spec/subaccounts.speq.js]: Damn, nice catch. I didn't even notice that. I blame our product. Will fix in a couple minutes.

File: docs/resources/subaccounts.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  Retrieve details about a specified subaccount by its id
+  * `subaccountId` - the id of the subaccount you want to look up **required**
+  * `callback` - see all function
+* **create(options, callback)**
+  Create a new subaccount
+  * `options.name` - user-friendly name **required**
+  * `options.keyLabel` - user-friendly identifier for subaccount API key **required**
+  * `options.keyGrants` - list of grants to give the subaccount API key **required**
+  * `options.keyValidIps` - list of IPs the subaccount may be used from
+  * `options.ipPool` - id of the default IP pool assigned to subaccount's transmissions
```
Comment (bizob2828) [inline docs/resources/subaccounts.md]: this can be set but has no effect yet.  This feature is coming soon

Comment (coldacid) [inline docs/resources/subaccounts.md]: It's still documented for the endpoint, therefore included.

**Comment by bizob2828** [comment]
Thanks again for the contribution.

**Pull request merged**